SillyBusinessInc/Moldbreaker
# Feat/#254 brute enemy

## Purpose of this PR:
Implement a Brute enemy that does double damage, and is a bit slower than the MeleeEnemy.
This takes the behaviours and statemachine from the MeleeEnemy without any changes. 
Changes were made to the MeleeEnemy.cs script to allow the Designers to select what type of attack an enemy does. This was to allow an easy and configurable way of setting which attack clip has to play when a MeleeEnemy attacks.
Changes were made to the MeleeAttackingState to incorporate above changes, so it's no longer hardcoded to clip 0 (stab) but now takes from the Enum in MeleeEnemy.cs

## How to test:
In any scene with the player and a navmeshsurface (so any level you get through starting a new game) place down a Brute enemy prefab ( `Assets/Production/Prefabs/Brute Enemy.prefab` ).
Confirm it does the brute slash spinning attack, confirm it has all of the other mechanics and behaviors from the MeleeEnemy.
this includes stuff like working healthbar, OnHit-flash when hit, roaming and idle behavior untill player comes in chase range, and more.

### links: 
closes #254
